bin/desi_job_graph

```diff
@@ -0,0 +1,147 @@
+
+p = argparse.ArgumentParser()
+p.add_argument('-n', '--night', type=int, required=True,
+        help='night to process')
+p.add_argument('-o', '--output', type=str, required=False,
+        help='output HTML file (default run/jobgraph/jobgraph-NIGHT.html in reduxdir)')
+p.add_argument('-s', '--specprod', type=str, required=False,
+        help=('override $SPECPROD, or full path '
+              'to override $DESI_SPECTRO_REDUX/$SPECPROD'))
+
+args = p.parse_args()
+
+#- Default output file into specprod directory
+if args.specprod is not None and os.path.isdir(args.specprod):
+    reduxdir = args.specprod
+else:
+    reduxdir = specprod_root(args.specprod)
+
+#- override environment variables for get_processing_table_* calls later
+os.environ['DESI_SPECTRO_REDUX'] = os.path.dirname(reduxdir)
+os.environ['SPECPROD'] = os.path.basename(reduxdir)
+
+if args.output is None:
+    args.output = f'{reduxdir}/run/jobgraph/jobgraph-{args.night}.html'
+    os.makedirs(os.path.dirname(args.output), exist_ok=True)
+
+#- avoid logging from get_processing_table...
+os.environ['DESI_LOGLEVEL'] = 'warning'
+
+#- Read processing table
+procpath = get_processing_table_path(None)
+procfile = get_processing_table_name(None, str(args.night))
+procfile = os.path.join(procpath, procfile)
+proctable = Table.read(procfile)
+
+#- Get the job state from rows that are actually slurm jobs
+ii = proctable['LATEST_QID']>0
+qinfo = queue_info_from_qids(proctable['LATEST_QID'][ii])
+jobinfo = {row['JOBID']:row for row in qinfo}
+
+#- Map production directory -> URL for log links
+reduxurl = reduxdir.replace(os.environ['DESI_ROOT'], 'https://data.desi.lbl.gov/desi')
+outputurl = args.output.replace(os.environ['DESI_ROOT'], 'https://data.desi.lbl.gov/desi')
+
+#- Write the mermaid graph
+fx = open(args.output, 'w')
+fx.write(f"""<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <style>
+    h1 {{font-family: Helvetica, sans-serif; margin-bottom: 0.1em;}}
+    p {{font-family: Menlo, Monospace; margin-top: 0.1em;}}
+  </style>
+</head>
+<body>
+  <h1>Night {args.night}</h1>
+  <p>{reduxdir}</p>
+  <div class="mermaid">
+  graph TD
+    classDef COMPLETED fill:#a8ddb5;
+    classDef PENDING   fill:#bdbdbd;
+    classDef RUNNING   fill:#bdd7e7;
+    classDef FAILED        fill:#d95f0e;
+    classDef OUT_OF_MEMORY fill:#d95f0e;
+    classDef TIMEOUT       fill:#d95f0e;
+    classDef CANCELLED     fill:#fed98e;
+    classDef UNKNOWN       fill:#ffffcc;
+""")
+
+jobs = dict()
+for row in proctable:
+    qid = row['LATEST_QID']
+    intid = row['INTID']        #- internal ID
+    jobdesc = row['JOBDESC']
+    expid = row['EXPID'][0:-1].replace('|', ',')  #- comma separated expids
+    e1 = int(expid.split(',')[0])  #- the first expid
+    tileid = row['TILEID']
+    proccamword = row['PROCCAMWORD']
+
+    if qid in jobinfo:
+        state = jobinfo[qid]['STATE'].split()[0]
+    else:
+        state = 'UNKNOWN'
+
+    if jobdesc == 'tilenight':
```

Comment:
I think this if-else will fail for redshift jobs that aren't stored under` scripts/night/` and have a different naming scheme.

Reply:
indeed, good catch; I updated this using `desispec.scripts.tile_redshifts.get_tile_redshift_script_pathname` (plus slurm -> log, adding qid...).